lib/containers/runtime.pm

```diff
@@ -0,0 +1,77 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Maintainer: qac team <qa-c@suse.de>
+package containers::runtime;
+use Mojo::Base qw(consoletest);
+use testapi;
+use Test::Assert ':all';
+
+sub new{
+    my ($class, $runtime) = @_;
+    my $self = {
+	runtime => $runtime,
+    };
+    bless $self, $class;
+    return $self;
+}
+
+sub build {
+    my ($self, $dockerfile_path, $container_tag) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    assert_script_run "cd $dockerfile_path";
+    assert_script_run "$self->{runtime} build -t $container_tag .";
+    record_info "$container_tag created";
+}
+
+sub up {
+    my ($self, $tag, $cmd, $daemon) = @_;
+    die 'tag is required' unless $tag;
+    my $mode = $daemon ? '-d' : '-it';
+    my $remote = $cmd ? 'sh -c' : '';
+    my $ret = script_run sprintf qq($self->{runtime} run --rm %s %s %s '%s'), $mode, $tag, $remote, $cmd;
+    record_info "Remote run on $tag", $cmd;
+    return $ret;
+}
+
+sub pull {
+    my ($self, $tag) = @_;
+    my $ret = script_run "$self->{runtime} pull $tag";
+    return $ret;
+}
+
+sub retrieve_images {
+    my ($self) = shift;
+    my %images = script_output qq/docker images | sed -n '1d;p' | awk '{print $1=$3}'/ =~ /^(\S+)="?([^"\r\n]+)"?$/gm;
+    return \%images;
+}
+
+sub retrieve_containers {
+    my ($self) = shift;
+    my %containers = script_output qq/docker container ls | sed -n '1d;p' | awk '{print $7=$1 }'/ =~ /^(\S+)="?([^"\r\n]+)"?$/gm;
```

Comment:
Perhaps we could do `docker container ls -q` instead of the `sed | awk` chain?

```suggestion
    my %containers = script_output qq/docker container ls -q/ =~ /^(\S+)="?([^"\r\n]+)"?$/gm;
```